lms/services/oauth_http.py

```diff
@@ -94,8 +103,23 @@ def refresh_access_token(self, token_url, redirect_uri, auth):
         :raise ExternalRequestError: if the HTTP request fails
         :raise ValidationError: if the server's access token response is invalid
         """
+        old_token = self._oauth2_token_service.get(self.service)
+
+        # If the "old" token is already current, just return immediately.
+        if (
+            old_token.access_token
+            and datetime.utcnow() - old_token.received_at < timedelta(seconds=30)
+        ):
+            return old_token.access_token
```

Comment:
I think it would make sense to use the same refresh behavior across all of our OAuth integrations if possible, to avoid the risks that come with rarely exercised code, and possible also reduce how often unnecessary reauthorizations occur. That was the rationale for having the logic centralized in `OAuthHTTPService`.

Reply:
Ah, I thought this was refusing to refresh a token if the token was not expired (according to our expiry time for it in our DB), but it's actually just "debouncing" (refusing to refresh a token if it was just refreshed in the last 30 seconds). That seems fine :+1: